sklearn/inspection/_partial_dependence.py

```diff
@@ -225,34 +226,24 @@ def partial_dependence(estimator, X, features, response_method='auto',
     method : str, optional (default='auto')
         The method used to calculate the averaged predictions:
 
-        - 'recursion' is only supported for gradient boosting estimator (namely
-          :class:`GradientBoostingClassifier<sklearn.ensemble.GradientBoostingClassifier>`,
-          :class:`GradientBoostingRegressor<sklearn.ensemble.GradientBoostingRegressor>`,
-          :class:`HistGradientBoostingClassifier<sklearn.ensemble.HistGradientBoostingClassifier>`,
-          :class:`HistGradientBoostingRegressor<sklearn.ensemble.HistGradientBoostingRegressor>`)
+        - 'recursion' is only supported for some tree-based estimators (namely
+          :class:`~sklearn.ensemble.GradientBoostingClassifier`,
+          :class:`~sklearn.ensemble.GradientBoostingRegressor`,
+          :class:`~sklearn.ensemble.HistGradientBoostingClassifier`,
+          :class:`~sklearn.ensemble.HistGradientBoostingRegressor`)
           but is more efficient in terms of speed.
-          With this method, ``X`` is only used to build the
-          grid and the partial dependences are computed using the training
-          data. This method does not account for the ``init`` predictor of
-          the boosting process, which may lead to incorrect values (see
-          warning below). With this method, the target response of a
+          With this method, the target response of a
           classifier is always the decision function, not the predicted
           probabilities.
 
         - 'brute' is supported for any estimator, but is more
           computationally intensive.
 
-        - 'auto':
+        - 'auto': the 'recursion' is used for estimators that support it,
+          and 'brute' is used otherwise.
 
-          - 'recursion' is used for
-            :class:`GradientBoostingClassifier<sklearn.ensemble.GradientBoostingClassifier>`
-            and
-            :class:`GradientBoostingRegressor<sklearn.ensemble.GradientBoostingRegressor>`
-            if ``init=None``, and for
-            :class:`HistGradientBoostingClassifier<sklearn.ensemble.HistGradientBoostingClassifier>`
-            and
-            :class:`HistGradientBoostingRegressor<sklearn.ensemble.HistGradientBoostingRegressor>`.
-          - 'brute' is used for all other estimators.
+        Please see :ref:`this note <pdp_method_differences>` for
+        differences between the 'brute' and 'recursion 'method.
```

Comment:
I would still keep a warning regarding the ``init`` estimator. IMO, this is something that is beyond just a difference between recursion and brute. We could use a `.. warning::` add have something succinct about it.